### PR TITLE
Don't bury killed buffers after counsel-switch-buffer

### PR DESCRIPTION
**Pull request by flatwhatson**
It's possible for a killed buffer to remain on buffer-list, causing bury-buffer to error while restoring the original buffer-list.  Erroring here leaves `counsel--switch-buffer-temporary-buffers` and `counsel--switch-buffer-previous-buffers` set with old values, leading to abnormal behavior on subsequent calls to `counsel-switch-buffer`.

### CODE REVIEW

File: counsel.el [hunk 1]
```diff
@@ -5058,8 +5058,11 @@ When ARG is non-nil, ignore NoDisplay property in *.desktop files."
 (defun counsel--switch-buffer-unwind ()
   "Clear temporary file buffers and restore `buffer-list'.
 The buffers are those opened during a session of `counsel-switch-buffer'."
-  (mapc 'kill-buffer counsel--switch-buffer-temporary-buffers)
-  (mapc 'bury-buffer counsel--switch-buffer-previous-buffers)
+  (dolist (buf counsel--switch-buffer-temporary-buffers)
+    (kill-buffer buf))
```
Comment (basil-conto) [inline counsel.el]: Why did you change the `mapc` to a `dolist`?

Comment (flatwhatson) [reply counsel.el]: Purely aesthetic, keeping similar things similar.  Maybe it would be better as:

    (mapc 'kill-buffer counsel--switch-buffer-temporary-buffers)
    (mapc 'bury-buffer (cl-remove-if-not
                        'buffer-live-p
                        counsel--switch-buffer-previous-buffers))

Comment (basil-conto) [reply counsel.el]: Bonus points for properly quoting function symbols as `#'kill-buffer`, `#'bury-buffer`, etc. ;)

Comment (flatwhatson) [reply counsel.el]: Done and done!

**Comment by flatwhatson** [comment]
@abo-abo Please let me know if you need anything else from me for this one.

**Comment by abo-abo** [comment]
Thanks.

**Comment by flatwhatson** [comment]
Thanks :heart:
